aspnetcore/security/blazor/server-side.md

```diff
@@ -0,0 +1,378 @@
+
+Server-side memory demands are a consideration for all server-side apps. However, most web apps are stateless, and the memory used while processing a request is released when the response is returned. As a general recommendation, don't permit clients to allocate an unbound amount of memory as in any other server-side app that persists client connections. The memory consumed by a Blazor server-side app persists for a longer time than a single request.
+
+> [!NOTE]
+> During development, a profiler can be used or a trace captured to assess memory demands of clients. A profiler or trace won't capture the memory allocated to a specific client. To capture the memory use of a specific client during development, capture a dump and examine the memory demand of all the objects rooted at a user's circuit.
+
+### Client connections
+
+Connection exhaustion can occur when one or more clients open too many concurrent connections to the server, preventing other clients from establishing new connections.
+
+Blazor clients establish a single connection per session and keep the connection open for as long as the browser window is open. The demands on the server of maintaining all of the connections isn't specific to Blazor apps. Given the persistent nature of the connections and the stateful nature of Blazor server-side apps, connection exhaustion is a greater risk to availability of the app.
+
+<!--
+
+For ...
+
+> Blazor clients establish a single connection per session and keep the connection open for as long as the browser window is open.
+
+It seems like the word "connection" is overloaded. The topic speaks earlier to multiple connections per circuit, but here there is only one "connection" per user (circuit). I also wonder about "browser window" versus "broswer tab" in this context. Should this read something like ...
+
+> Blazor clients can establish multiple connections per user (circuit) and keep the connections open for as long as the browser window is open.
+
+-->
+
+By default, there's no limit on the number of connections per user for a Blazor server-side app. If the app requires a connection limit, take one or more of the following approaches:
+
+* Require authentication, which naturally limits the ability of unauthorized users to connect to the app. For this scenario to be effective, users must be prevented from provisioning new users at will.
+* Limit the number of connections per user. Limiting connections can be accomplished via the following approaches. Exercise care to allow legitimate users to access the app (for example, when a connection limit is established based on the client's IP address).
+  * At the app level:
+    * Endpoint routing extensibility.
+    * Require authentication to connect to the app and keep track of the active sessions per user.
+    * Reject new sessions upon reaching a limit.
```

Comment:
Do we really think it's reasonable for people to implement their own connection limits? This seems like something that would be really hard to do correctly